### DISCUSSION TITLE
how can I get rid of beeping forever?

### DISCUSSION THREAD

**Discussion by catharsis71** [microsoft/WSL#8680] [Q&A]
it's 2022 we still live in a world where if you accidentally cat a large binary file you're in for hours of audio torment

many ostensible "solutions" were found online but none of them work for me

I am not using Windows Terminal
I do not have a "Console Window Host" in my volume mixer or "app volume and device preferences"
Editing /etc/inputrc gets rid of some beeping but not all
"Critical Stop" system sound is already set to (None)
"setterm -blength 0" doesn't work ("setterm: terminal xterm-256color does not support --blength")
muting "System Sounds" entirely does work but I don't want to do that (overkill/collateral damage)
I have gone through the System Sounds hitting Test on every item that has a sound assigned and none of them match "my" beep
If I change the sound scheme to Windows Default the beeping is replaced with a much worse beep
I apparently set a custom sound scheme years ago where I turned off many of the sounds or replaced them with less-annoying versions but the beep I'm hearing doesn't seem to be on there
Even browsing through the WAVs in c:\windows\media I can't find the beep I'm hearing
Changing the sound scheme to "No Sounds" does work but I don't want to do that (overkill/collateral damage)
The audio is definitely coming from the TV that I use as a monitor & not from my PC speaker; the PC is on the opposite side of the room and also doesn't have a PC speaker installed

I know that terminal beeps have been a plague on society for more than 50 years, before the advent of WSL, but I hope WSL can offer some kind of solution.

**Comment by catharsis71** [comment]
Doing further research, I found ```net stop beep``` as a proposed solution.  Initial results seemed not great...

```
C:\WINDOWS\system32>net stop beep
The Beep service is stopping........
The Beep service could not be stopped.


C:\WINDOWS\system32>
C:\WINDOWS\system32>
C:\WINDOWS\system32>net stop beep
The service is starting or stopping.  Please try again later.


C:\WINDOWS\system32>net stop beep
The service is starting or stopping.  Please try again later.


C:\WINDOWS\system32>
```

and yet, the beeping has stopped!  for now....

but I'm not sure of possible collateral damage, and I also can't find this service in services.msc in order to disable it permanently

**Reply by DanKaplanSES** [reply]
I want to make sure I understand what you achieved: you ran a command in an elevated `cmd.exe`, and this turned off the beeps in your wsl terminal?

**Comment by NotTheDr01ds** [comment]
> and yet, the beeping has stopped! for now

Well that's encouraging, at least.

> `The Beep service could not be stopped.`

I'm seeing the same thing on Windows 11, then the Service goes into `StopPending` state and stays there.

> I also can't find this service in services.msc

Now that took me a bit to figure out.  It turns out it isn't actually a "Service", but it's really a "Device".  In PowerShell:

```PowerShell
$ Get-Service | Get-Member -static

   TypeName: System.ServiceProcess.ServiceController
Name            MemberType Definition
----            ---------- ----------
Equals          Method     static bool Equals(System.Object objA, System.Object objB)
GetDevices      Method     static System.ServiceProcess.ServiceController[] GetDevices(), static System.ServiceProcess…
GetServices     Method     static System.ServiceProcess.ServiceController[] GetServices(), static System.ServiceProces…
...

$ [System.ServiceProcess.ServiceController]::GetDevices() |? { $_.Name -eq "Beep" }

Status   Name               DisplayName
------   ----               -----------
Stopped  Beep               Beep
```

> in order to disable it permanently

Even though it's a device, you can still change its startup type through PowerShell (must be Admin):

```PowerShell
Set-Service Beep -StartupType Disabled
```

Reboot, and it should be disabled permanently.

According to [this Reddit thread](https://www.reddit.com/r/Windows10/comments/6dv1on/disabling_all_sounds_in_powershell/?utm_source=share&utm_medium=web2x&context=3), it's also possible to do the same in an Admin CMD session with:

```
sc.exe config beep start= disabled
```

**Comment by NotTheDr01ds** [comment]
Side note:  Setting the beep device to start automatically seems to only be "correctly" done from an administrative CMD shell, since PowerShell doesn't allow the StartupType of `System` (the original setting).

```
sc.exe config beep start= System
```

It's possible to set it to `Automatic` from PowerShell, but the original value was `System`, for whatever it's worth.
